lib/web/app/handler.go

```diff
@@ -128,8 +129,8 @@ func NewHandler(ctx context.Context, c *HandlerConfig) (*Handler, error) {
 	// Create the application routes.
 	h.router = httprouter.New()
 	h.router.UseRawPath = true
-	h.router.POST("/x-teleport-auth", makeRouterHandler(h.withCustomCORS(h.handleAuth)))
-	h.router.OPTIONS("/x-teleport-auth", makeRouterHandler(h.withCustomCORS(nil)))
+	h.router.GET("/x-teleport-auth", makeRouterHandler(h.startAppAuthExchange))
+	h.router.POST("/x-teleport-auth", makeRouterHandler(h.completeAppAuthExchange))
```

Comment:
Do we need to keep the `OPTIONS` call available for a bit?

As written, app access may break during upgrades when not all proxies are on the same version.

Reply:
how long will proxy versions differ though? (when ryan's changes went it, the app breakage wasn't taken into account) 

if we were to handle preventing temporary breakage, I would need to make `POST /x-teleport-auth` differentiate between `cors request with custom headers` and `non-cors request with state query param and auth cookie`

Reply:
i added backwards compat starting from this commit: https://github.com/gravitational/teleport/pull/30321/commits/cc691d09f03b9261edd32854f96984cdda6d433b

tested with v15 build using new and legacy AppLauncher.tsx component with `https://grafana-test-127-0-0-1.nip.io:3080/alerting/list?search=state:inactive%20type:alerting%20health:nodata`